core/domain/rte_component_registry_test.py

```diff
@@ -67,11 +67,19 @@ def _validate_customization_arg_specs(self, customization_arg_specs):
             self.assertTrue(isinstance(ca_spec['description'], basestring))
             self.assertGreater(len(ca_spec['description']), 0)
 
+            schema_utils_test.validate_schema(ca_spec['schema'])
+            # Default value of SanitizedUrl obj_type may be empty. Empty value
+            # cannot normalized, so we don't normalize it.
+            if ca_spec['schema']['type'] == 'custom' and (
+                    ca_spec['schema']['obj_type'] == 'SanitizedUrl'):
+                if ca_spec['default_value'] == '':
+                    self.assertEqual(len(ca_spec['default_value']), 0)
+                    return
+
             # The default value might not pass validation checks (e.g. the
```

Comment:
So I just read this comment, which makes it sound like this case you're trying to address here was already accounted for in previous code.

Should we just move the nonempty check to a "custom validator" in the SanitizedUrl object?

Reply:
To add nonempty check "custom validator" in SanitizedUrl object, I will have have to allow empty values in `sanitize_url` [here](https://github.com/oppia/oppia/blob/develop/schema_utils.py#L211). Should I proceed with this approach?

Reply:
Sorry, I don't understand. What changes exactly are you planning to make to sanitize_url?

Reply:
Currently `sanitize_url` gives error if empty string is passed to it. So I plan to modify it so that empty strings can pass through `sanitize_url`. After that I will add a "custom validator" to ensure that string is non-empty. This will ensure that default value which is an empty string can be normalized using `sanitize_url`.

Reply:
Just to be clear, what error does it give? I don't see a "raise" in the function or a Raises section in the docstring.

I think the general suggestion should be fine, but make sure to modify the docstring and add tests to confirm the new behaviour.

Reply:
Earlier tests were creating problem because passing empty string to `sanitize_url` gave error [here](https://github.com/oppia/oppia/blob/develop/schema_utils.py#L227). To resolve this, I have allowed empty strings to pass through `sanitize_url` and added custom validator to check non-empty strings in SanitizedUrl object.